src/test/java/seedu/address/logic/commands/ClientEditCommandTest.java

```diff
@@ -156,6 +171,34 @@ public void execute_invalidClientIndexFilteredList_failure() {
         assertCommandFailure(clientEditCommand, model, Messages.MESSAGE_INVALID_CLIENT_DISPLAYED_INDEX);
     }
 
+    // Editing a client's non-note related fields should retain existing notes.
+    @Test
+    public void execute_changeClientName_preserveExistingClientNotes() throws CommandException {
+        Note clientNote = new Note("this note needs to persist after client gets edited");
+        Set<Tag> tags = new HashSet<>();
+        tags.add(new Tag("niceTag"));
+        clientNote.setTags(tags);
+        Index indexLastClient = Index.fromOneBased(model.getSortedFilteredClientList().size());
+        Client lastClient = model.getSortedFilteredClientList().get(indexLastClient.getZeroBased());
+        model.addClientNote(lastClient, clientNote);
+        assertTrue(model.getSortedFilteredClientNotesList().size() == 1);
+        ClientBuilder clientInList = new ClientBuilder(lastClient); // rename to currentClient
+        Client editedClient = clientInList.withName(VALID_NAME_BOB).withPhone(VALID_PHONE_BOB).build();
+        editedClient.addClientNote(clientNote);
+        EditClientDescriptor descriptor = new EditClientDescriptorBuilder().withName(VALID_NAME_BOB)
+                .withPhone(VALID_PHONE_BOB).build();
+        ClientEditCommand clientEditCommand = new ClientEditCommand(indexLastClient, descriptor);
+        String expectedMessage = String.format(ClientEditCommand.MESSAGE_EDIT_CLIENT_SUCCESS, editedClient);
+        Model expectedModel = new ModelManager(new TbmManager(model.getTbmManager()), new UserPrefs());
+        expectedModel.setClient(lastClient, editedClient);
```

Comment:
shouldn't setClient method in modelManager also update the tagNoteMap? Not sure where its being done currently.